roosevelt.js

```diff
@@ -21,16 +23,59 @@ module.exports = function(params) {
     }
   });
   
-  var app = express(),                // initialize express
-      httpServer = http.Server(app);  // create http server out of the express app
+  var app = express(), // initialize express
+      httpServer,
+      httpsServer;
 
   // expose initial vars
   app.set('express', express);
   app.set('params', params);
-  app.httpServer = httpServer;
   
   // source user supplied params
   app = require('./lib/sourceParams')(app);
+  
+  // let's try setting up the servers with user-supplied params
+  if (!app.get('params').httpsOnly) {
+    httpServer = http.Server(app);
+  }
+  
+  if (app.get('params').https) {
+    var httpsOptions = {
+          requestCert: app.get('params').requestCert,
+          rejectUnauthorized: app.get('params').rejectUnauthorized
+        },
+        ca = app.get('params').ca,
+        passphrase = app.get('params').passphrase;
+
+    if (app.get('params').keyPath) {
+      if (app.get('params').pfx) {
+        httpsOptions.pfx = fs.readFileSync(app.get('params').keyPath.pfx);
+      }
+      else {
+        httpsOptions.key = fs.readFileSync(app.get('params').keyPath.key);
+        httpsOptions.cert = fs.readFileSync(app.get('params').keyPath.cert);
+      }
+      if (passphrase) {
+        httpsOptions.passphrase = passphrase;
+      }
+      if (ca) {
+        // String or array
+        if (typeof ca === String) {
+          httpsOptions.ca = fs.readFileSync(ca);
+        }
+        else if (typeof ca === Array) {
```

Comment:
`typeof` only recognizes primitive types and an `Object`. You can use the `instanceof` operator to compare the variable with the `Array` object to test the variables prototype chain against Array (i.e. `ca instanceof Array`). It outputs a boolean.